api/task_status.go

```diff
@@ -0,0 +1,155 @@
+package api
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"strings"
+
+	"github.com/hashicorp/consul-terraform-sync/event"
+)
+
+const taskStatusPath = "status/tasks"
+
+// TaskStatus is the status for a single task
+type TaskStatus struct {
+	TaskName  string   `json:"task_name"`
+	Status    string   `json:"status"`
+	Providers []string `json:"providers"`
+	Services  []string `json:"services"`
+	EventsURL string   `json:"events_url"`
+}
+
+// taskStatusHandler handles the task status endpoint
+type taskStatusHandler struct {
+	store   *event.Store
+	version string
+}
+
+// newTaskStatusHandler returns a new TaskStatusHandler
+func newTaskStatusHandler(store *event.Store, version string) *taskStatusHandler {
+	return &taskStatusHandler{
+		store:   store,
+		version: version,
+	}
+}
+
+// ServeHTTP serves the task status endpoint which returns a map of taskname to
+// task status
+func (h *taskStatusHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	log.Printf("[TRACE] (api.taskstatus) requesting task status '%s'", r.URL.Path)
```

Comment:
nit to rename as `api.status.tasks` to be more predictable with the API endpoint?

Reply:
I like this idea to have it align with the API endpoint. It does sort of breakaway from the other logs which seems to be approximately `<package-name>.<file/object name>`. It could be confusing. Maybe `(api./status/tasks)`? The slashes might make it more clear that it's not a hierarchy and visually look like the api endpoint? Let me know if you have other ideas

Reply:
Agree with not mixing delimiters! All right, I'll leave it as is for now. We've talked about switching to structured logs - we can revisit this when we do that. Let me know if your thoughts change.